vpp-patches/0034-fix-don-t-rename-nodes-of-tunnel-interfaces.patch

```diff
@@ -0,0 +1,155 @@
++++ b/src/plugins/gre/gre.c
+@@ -763,6 +763,7 @@ VNET_DEVICE_CLASS (gre_device_class) = {
+   .format_tx_trace = format_gre_tx_trace,
+   .admin_up_down_function = gre_interface_admin_up_down,
+   .ip_tun_desc = gre_tunnel_desc,
++  .has_common_nodes = 1,
+ #ifdef SOON
+   .clear counter = 0;
+ #endif
+diff --git a/src/plugins/gtpu/gtpu.c b/src/plugins/gtpu/gtpu.c
+index 1307794b9..499ac7e7f 100644
+--- a/src/plugins/gtpu/gtpu.c
++++ b/src/plugins/gtpu/gtpu.c
+@@ -160,6 +160,7 @@ VNET_DEVICE_CLASS (gtpu_device_class,static) = {
+   .format_device_name = format_gtpu_name,
+   .format_tx_trace = format_gtpu_encap_trace,
+   .admin_up_down_function = gtpu_interface_admin_up_down,
++  .has_common_nodes = 1,
+ };
+ /* *INDENT-ON* */
+ 
+diff --git a/src/plugins/l2tp/l2tp.c b/src/plugins/l2tp/l2tp.c
+index a975fa7f0..6a9be4510 100644
+--- a/src/plugins/l2tp/l2tp.c
++++ b/src/plugins/l2tp/l2tp.c
+@@ -259,6 +259,7 @@ VNET_DEVICE_CLASS (l2tpv3_device_class,static) = {
+   .name = "L2TPv3",
+   .format_device_name = format_l2tpv3_name,
+   .name_renumber = l2tpv3_name_renumber,
++  .has_common_nodes = 1,
+ };
+ /* *INDENT-ON* */
+ 
+diff --git a/src/plugins/vxlan/vxlan.c b/src/plugins/vxlan/vxlan.c
+index f1ab7a7cb..668fa4274 100644
+--- a/src/plugins/vxlan/vxlan.c
++++ b/src/plugins/vxlan/vxlan.c
+@@ -134,6 +134,7 @@ VNET_DEVICE_CLASS (vxlan_device_class, static) = {
+   .format_device_name = format_vxlan_name,
+   .format_tx_trace = format_vxlan_encap_trace,
+   .admin_up_down_function = vxlan_interface_admin_up_down,
++  .has_common_nodes = 1,
+ };
+ /* *INDENT-ON* */
+ 
+diff --git a/src/vnet/interface.c b/src/vnet/interface.c
+index db47d42c8..7bd679d84 100644
+--- a/src/vnet/interface.c
++++ b/src/vnet/interface.c
+@@ -1149,9 +1149,9 @@ vnet_delete_hw_interface (vnet_main_t * vnm, u32 hw_if_index)
+ 	}
+ 
+       vlib_node_rename (vm, hw->output_node_index,
+-			"interface-%d-output-deleted", hw_if_index);
+-      vlib_node_rename (vm, hw->tx_node_index, "interface-%d-tx-deleted",
+-			hw_if_index);
++			"interface-%d-%d-output-deleted", hw_if_index, hw->output_node_index);
++      vlib_node_rename (vm, hw->tx_node_index, "interface-%d-%d-tx-deleted",
++			hw_if_index, hw->tx_node_index);
+       vlib_unregister_errors (vm, hw->output_node_index);
+       vlib_unregister_errors (vm, hw->tx_node_index);
+       vec_add2 (im->deleted_hw_interface_nodes, dn, 1);
+@@ -1577,9 +1577,16 @@ vnet_rename_interface (vnet_main_t * vnm, u32 hw_if_index, char *new_name)
+   hash_unset_mem (im->hw_interface_by_name, old_name);
+   hash_set_mem (im->hw_interface_by_name, hw->name, hw_if_index);
+ 
+-  /* rename tx/output nodes */
+-  vlib_node_rename (vm, hw->tx_node_index, "%v-tx", hw->name);
+-  vlib_node_rename (vm, hw->output_node_index, "%v-output", hw->name);
++  vnet_device_class_t *dev_class =
++    vnet_get_device_class (vnm, hw->dev_class_index);
++
++  /* only rename nodes of non-tunnel ifaces */
++  if (!dev_class->has_common_nodes)
```

Comment:
Seems like this check should replace 
```
if (dev_class->tx_function)
```
in `vnet_delete_hw_interface`. To avoid entering reuse pool at all.